src/error.rs

```diff
@@ -0,0 +1,33 @@
+use std::error::Error;
+use std::fmt;
+
+#[derive(Debug)]
+pub struct InfoError {
+    details: String,
+}
+
+impl InfoError {
+    pub fn new(msg: &str) -> InfoError {
+        InfoError {
+            details: msg.to_string(),
+        }
+    }
+}
+
+impl fmt::Display for InfoError {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        write!(f, "{}", self.details)
+    }
+}
+
+impl Error for InfoError {
+    fn description(&self) -> &str {
+        &self.details
+    }
+}
+
+impl From<goblin::error::Error> for InfoError {
+    fn from(err: goblin::error::Error) -> Self {
+        InfoError::new(&err.to_string())
```

Comment:
if you go with my suggestion above, the `to_string` here can go away (also the reference)